src/modules/verifiable-credentials/verifiable-credentials-base.service.ts

```diff
@@ -522,6 +530,10 @@ export abstract class VerifiableCredentialsServiceBase {
     return pex.selectFrom(presentationDefinition, vc);
   }
 
+  public filterSelfSignDescriptors(descriptors) {
```

Comment:
@jrhender Sure thing, but I don't think we can test private methods in e2e test files - if this is the case, should I just remove the test? I don't see any other private methods in e2e tests so I dont think you can import them but could be wrong.

Reply:
I will move it to a util file and test it there.

Reply:
@whitneypurdum Right, good point, we probably can't test private methods. However, IMO, unit tests should test the public API of a class because testing private methods is a code (test?) smell that the tests are testing implementation details (which means that the tests need to change if the implementation details change, which somewhat defeats the purpose of tests).
I think the `util` file is actually worse then just making it public as I think util implies that it is a general function (which it isn't really IMO) and moves it away from the place where it is being used.

IMO, a better approach in this case would be to use a `spy` (e.g. https://jestjs.io/docs/jest-object#jestspyonobject-methodname) to spy on pex and see that we are passing the expected object to it. I'd grant that this is still testing an implementation detail, but it is more testing an aspect of the implementation and we could check the way that the implementation is accomplishing this aspect without breaking the test.
What are your thoughts on this?

If the above is too complicated, we could just stick with keeping it public, no problem. Let me know what you think.

Reply:
Sure thing, I will implement the spy then.

Reply:
Ok, hopefully the `spy` is a good suggestion 😆 . If you start using it and it doesn't work, let's discuss